optimum/onnxruntime/modeling_ort.py

```diff
@@ -445,18 +450,40 @@ def forward(
         token_type_ids: Optional[torch.Tensor] = None,
         **kwargs,
     ):
-        # converts pytorch inputs into numpy inputs for onnx
-        onnx_inputs = {
-            "input_ids": input_ids.cpu().detach().numpy(),
-            "attention_mask": attention_mask.cpu().detach().numpy(),
-        }
-        if token_type_ids is not None:
-            onnx_inputs["token_type_ids"] = token_type_ids.cpu().detach().numpy()
-        # run inference
-        outputs = self.model.run(None, onnx_inputs)
-        last_hidden_state = torch.from_numpy(outputs[self.model_outputs["last_hidden_state"]]).to(self.device)
-        # converts output to namedtuple for pipelines post-processing
-        return BaseModelOutput(last_hidden_state=last_hidden_state)
+        if self.device.type == "cuda" and self.use_io_binding:
+            onnx_inputs = {"input_ids": input_ids, "attention_mask": attention_mask}
+            if token_type_ids is not None:
+                onnx_inputs["token_type_ids"] = token_type_ids
+            io_helper = IOBindingHelper(self.model, self.config, self.device)
```

Comment:
@philschmid actually I placed it here because it will throw an error if I place it in the `__init__`

```
Traceback (most recent call last):
  File "test_ort_infer_iobinding.py", line 104, in <module>
    output = v_onnx_model(**d)
  File "/workspace/optimum/modeling_base.py", line 57, in __call__
    return self.forward(*args, **kwargs)
  File "/workspace/optimum/onnxruntime/modeling_ort.py", line 1000, in forward
    io_binding = self.io_helper.prepare_io_binding(**onnx_inputs)
  File "/workspace/optimum/onnxruntime/io_binding/io_binding_helper.py", line 108, in prepare_io_binding
    io_binding.bind_input(
  File "/usr/local/lib/python3.8/dist-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 458, in bind_input
    C.OrtDevice(
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. onnxruntime.capi.onnxruntime_pybind11_state.OrtDevice(arg0: int, arg1: int, arg2: int)
```

Reply:
Are you sure that's coming from placing it in the `__init__`? and might not be and order issue?